src/PrestaShopBundle/Kernel/ModuleRepositoryFactory.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Kernel;
+
+
+use Doctrine\DBAL\DriverManager;
+
+class ModuleRepositoryFactory
+{
+    /**
+     * @var self
+     */
+    private static $instance;
+
+    /**
+     * @var string
+     */
+    private $parameters;
+
+    /**
+     * @var ModuleRepository
+     */
+    private $moduleRepository;
+
+    public static function getInstance()
```

Comment:
singleton anti-pattern spotted aka "omfg I can't figure out why my tests randomly failed!" drama

Reply:
yeah, I was not sure of this one, I just wanted to avoid building several `ModuleRepository` instances for performance issues, as this class is basically used in contexts where we don't have the symfony container available
I can drop all the singleton management, but then changes are we will create doublons of `ModulesRepository` and perform the query several times

Reply:
can you provide a with/without singleton report from Blackfire please?

Reply:
the difference is negligible, it doesn't have enough impact to consider one faster/slower than the other